### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by vincent-tr** [HDF-NI/hdf5.node#19]
Hi!

I saw several points that could make the code more readable, and maybe more efficient:

- We should not threat hpp file as source file (in binding.gyp). hpp/h file are headers, and should not be considered as sources.

- It may be more readable (and more efficient for the compiler) to have header files (h/hpp) that declare functions/classes prototypes and cc files that implement them. It is done in file.h/h5_file.cc, and group.h/h5_group.cc. IMHO it should be done for all other classes. I also think we should decide if we use .h or .hpp for haeder files, and use one unique convention.

- We may not use namespace in header files. It is generally a bad practice as it pollute the global namespace in a header, which is meant to be included in several source files, and can then cause names conflicts in source files.

- We should have proper code indentation. I think it could be a good idea to have also common code formatting.

Of course we can discuss about any point above if needed, this are only ideas.
If you agree on some of the above points, I can work on them, and propose pull requests.
Let me know.

**Comment by shumway** [comment]
@vincent-tr  and @rimmartin, thank you for posting these issues on a public channel. I would like to help out the project, and this helps me learn what's going on and how I might contribute. I've worked with HDF for over twenty years, and I am most experienced with C++ and Python. I have some experience with node and have worked on simple HDF5 addon bindings for internal projects. I'm looking to improve my node.js skills and contribute to an open source project. You can see my pi-qmc project for some examples of what I've done with continuous integration and documentation, though the source code and structure is uglier than what I prefer now. Please let me know how I can help.

**Comment by rimmartin** [comment]
Yes, very much in agreement. I had rapped this out in standalone cloud9 without good formatting; usually I use netbeans which has set formatting; and can customize it to match formatting everyone likes

Yea the name spacing came from copying examples and I don't like them in headers either. 

Wasn't sure if I would eventually template something so started going with hpp with everything in one file for  class.  Yet a unified style for this project is a good thing

And you are right about faster compile; although I was trying to precompile for some users.  yet npm doesn't yet allow for the entire matrix of builds to be published.  I never found a way to provide for multiple versions of the native hdf5 libraries

I am very encouraged with more developers!

**Comment by rimmartin** [comment]
https://github.com/nodejs/node-gyp/issues/893

Working with node gyp people on defaulting variables yet allowing user to set when the deault doesn't work for them

**Comment by rimmartin** [comment]
Hi @shumway, pi-qmc is some serious coding:-) 

Is there an aspect of hdf5 node that interests you?  Besides the core there are some layers to go.  There is the image API aspect. 

For my own use I have images from scientific instruments and also 3 rank matrices going to http://pointclouds.org/ based applications and 3d fourier transforms

In this interface the intention isn't to replicate the hdf5 API but to map and get the data as directly to javascript style as possible.

**Comment by shumway** [comment]
Hi @rimmartin. That pi-qmc project is fun, but essentially stalled since I left my professorship a few years ago. I've followed this nodejs hdf5 project for about a year, and I tried to use it recently for a command line utility and to see if it would work with our internal toolsets. I ended up writing some of my own simplified addon wrappers instead, specific to our application, and wrote a script using python/pytables. That said, I think node is very powerful, and hdf5.node is an important project. There isn't anything I need; I just want to give back to the community. It looks like the best way I can be helpful is to work on documentation, so I'm opening that up as a new issue to keep from further hijacking @vincent-tr 's code cleanup thread.

**Comment by rimmartin** [comment]
Hi @vincent-tr ,

I've got some fixes and additions for image io that I'm preparing to publish.  Is There anything you would like published to npm as well?

**Comment by vincent-tr** [comment]
Hi,

Sorry, nothing really ready for a publish right now :/

**Comment by rimmartin** [comment]
Oh, that's fine.  I just wanted to check so I don't publish twice days apart
